src/devhub/devhub.js

```diff
@@ -172,24 +175,45 @@ function batchesToSeries(batches) {
         series.unit = metric.unit;
       }
 
-      series.value.push(metric.value);
+      // Even though our x-axis is time, we want to spread things out evenly by batch, rather than
+      // group according to time. Apex charts is much quicker when given an x value, even though it
+      // isn't strictly needed.
+      series.value.push([index, metric.value]);
       series.git_commit.push(batch.attributes.git_commit);
       series.timestamp.push(batch.timestamp);
     }
   }
-  return result.values();
+
+  // Sort the results by batches that have the latest entries first, so metrics that have stopped
```

Comment:
Let’s do a little bit more here: sort series’s by their order of appearance in data.

that way, if we change the order of things in zig code for the latest batch, it’ll affect the ordering of the graphs.

this’ll naturally sort shorter series to the bottom.

impl wise, I’d imagine that we can do this online, without explicit sorting: when adding a new series, append its name to the list of series.

so, after processing the first batch we sort all current series

Reply:
Hm, do we even need any code here? JavaScript honors insertion order, no?

Reply:
> Hm, do we even need any code here? JavaScript honors insertion order, no?

It does, but because of how the data is structured it means early metrics that now disappeared will still be in their original position high up.

I'll make it so that it'll be ordered first by current batch metric ordering, then by appearance for old batches.

Reply:
That reverses the data too, so it needs [some tweaking](https://github.com/tigerbeetle/tigerbeetle/compare/71f753c98ecd74eaeada53f72eb4d92f7c3d2c86..087606f18d5578def44a7dc595d05ce3d18bee19) to render correctly.